src/Database/Expression/CaseExpression.php

```diff
@@ -130,6 +130,10 @@ protected function _addExpressions($conditions, $values, $types)
                 $value = $keyValues[$k];
                 array_push($this->_values, $value);
                 continue;
+            } elseif ($value === 'identifier') {
```

Comment:
Yeah that's probably out of scope of this PR I think.
But in general else should be avoided wherever possible. Compactness is somewhat irrelevant here.